File: redsheet-portal/src/app/database-models/application-object.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class ApplicationObject {
    constructor(applicationObject?: ApplicationObject )  {
        let properties = [ "applicationObjectId", "applicationObjectName", "applicationObjectTypeId",];
        for (let property of properties)
            if (applicationObject && applicationObject[property])
                this[property] = applicationObject[property];
    }
 
	applicationObjectId : number =   0 ;
 
    @required()
    @maxLength(100)
	applicationObjectName : string =   undefined;
 
    @range(1,2147483647)
	applicationObjectTypeId : number =   undefined;


}

File: RedSheet.DbEntities/Models/ApplicationObject.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	///<summary>
    ///Application Objects is used Application wide
    ///</summary>
	[Table("ApplicationObjects", Schema = "dbo")]
	[Serializable]
	public partial class ApplicationObject  
	{
	///<summary>
    ///Main Primary Key
    ///</summary> 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int ApplicationObjectId { get; set; } 
 
	[Required] 
	[MaxLength(100)]
	public string ApplicationObjectName { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ApplicationObjectTypeId { get; set; } 

	public ApplicationObject()
	{
    }

    }
}

